src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -974,7 +979,9 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
                     if !constant_arguments.contains(&i) {
                         return
                     }
-                    if this.qualif.is_empty() {
+                    // if the argument requires a constant, we care about constness, not
```

Comment:
This could do with elaboration. What do we mean by "requires a constant"? Where is this checked for?